Feat/hud

Xander-Allen-Rola opened the pull request:
Additions:

- Status Bars (Health, Hunger, Thirst)
- NPC Description Prompt

Xander-Allen-Rola committed:
added status bars HUD

Xander-Allen-Rola committed:
placed status bar hud in world scene

Xander-Allen-Rola committed:
Added fonts for prompt and placed status bars in world

Xander-Allen-Rola committed:
Merge branch 'dev' of https://github.com/janrizmlibres/emergent-echoes into feat/hud

Xander-Allen-Rola committed:
Merged develop with feat/hud branch

Xander-Allen-Rola committed:
placed status bars in world camera

Xander-Allen-Rola committed:
Added Interatction HUD with NPC

Xander-Allen-Rola committed:
NPC Description prompt now shows when player interacts with NPC

Xander-Allen-Rola committed:
Programmed scripts for interacting with SIlas to show NPC description. programmed methods for changing the contents of the prompt

Xander-Allen-Rola committed:
Added prompt interaction for garreth

Xander-Allen-Rola committed:
Merge branch 'dev' of https://github.com/janrizmlibres/emergent-echoes into feat/hud

Xander-Allen-Rola committed:
Fixed Interaction error with NPC prompts

Xander-Allen-Rola committed:
merge conflict from dev

janrizmlibres reviewed (changes requested):
Put interface-related stuff in UI folder under Entities

janrizmlibres commented on Utilities/IInteractable.cs:
Why the double I's? Seems like there are two Interactables. Consider another name.

Xander-Allen-Rola committed:
added heart bars for npc relationships

Xander-Allen-Rola committed:
Refactor NpcPrompt to be reusable and update derived classes

- Moved common functionality to NpcPrompt.cs in the interface folder
- Updated GarrethPrompt.cs and SilasPrompt.cs to inherit from NpcPrompt
- Changed DisplayDescription method in NpcPrompt to protected for accessibility

Xander-Allen-Rola committed:
Moved Status bar assets and scenes, fonts, and heart bar assets and scenes inside UI folder

Xander-Allen-Rola committed:
moved interface files to appropriate folders

Xander-Allen-Rola committed:
Rename IInteractable to InteractableObject
